### PR TITLE
Masking during finetuning

### PR DESCRIPTION
**Pull request by gabrieltseng**
@kvantricht , main questions:

1. Do we want to apply masking during validation too?
2. Which masking strategies do we want to use? Currently, we use all (which may make the model more robust to varied missing value types)

### CODE REVIEW

File: presto/dataset.py [hunk 1]
```diff
@@ -308,6 +319,10 @@ def __getitem__(self, idx):
         df_index = self.indices[idx]
         row = self.df.iloc[df_index, :]
         eo, mask_per_token, latlon, month, target = self.row_to_arrays(row, self.target_function)
+        if self.mask_ratio > 0:
+            mask_per_token, eo, _, _ = self.mask_params.mask_data(
+                self.normalize_and_mask(eo), mask_per_token
```
Comment (gabrieltseng) [inline presto/dataset.py:324]: Don't call normalize and mask here, it gets called later

**Comment by gabrieltseng** [comment]
Next steps:
- Gabi fixed bug he just found
- Merge into main following review from @cbutsko 
- main merged into crop type branch to include these changes

**Comment by kvantricht** [comment]
> @kvantricht , main questions:
> 
> 1. Do we want to apply masking during validation too?
> 2. Which masking strategies do we want to use? Currently, we use all (which may make the model more robust to varied missing value types)

1. In case the seed makes sure all experiments are intercomparable, then we could opt for it, but I'm also not sure. If we want to verify how different methods perform during validation in specific circumstances including masked channels, then we may benefit from it. Guess my main concern is comparability of the results.
2. I guess all of them indeed can make the model more robust, even though e.g. `random` is not likely to happen as such in reality. But probably also doesn't hurt.

**Comment by gabrieltseng** [comment]
@kvantricht awesome. So am I right in understanding that:

1. We will not apply masking during validation to keep results comparable
2. We will apply all masking strategies

We might want to update our validation set with some (fixed) masking so that it better reflects deployment conditions, but this is something we can do later

**Comment by kvantricht** [comment]
> @kvantricht awesome. So am I right in understanding that:
> 
> 1. We will not apply masking during validation to keep results comparable
> 2. We will apply all masking strategies
> 
> We might want to update our validation set with some (fixed) masking so that it better reflects deployment conditions, but this is something we can do later

Agreed! For that last point, we can plan to extract some inference patches that are more recent and capture a true absence of Sentinel-1 data. As you pointed out, this is easy to do along the way. We now have a nice workflow that easily adds patches to the list.

**Comment by gabrieltseng** [comment]
@kvantricht awesome - made that change. Let me know if this is good to merge

File: presto/dataset.py (hunk 1 above)
Comment (kvantricht) [reply presto/dataset.py:324]: Regarding this comment, did you double check that? `normalize_and_mask` still called on `eo` and when returning from the function `normalize_and_mask` called once again? This doesn't do double normalization?

**Comment by kvantricht** [comment]
> @kvantricht awesome - made that change. Let me know if this is good to merge

added one more question based on an earlier comment of yours.

File: presto/dataset.py (hunk 1 above)
Comment (gabrieltseng) [reply presto/dataset.py:324]: thanks for catching this! i left this comment intending to remove this call and forgot 🙈

Comment (gabrieltseng) [reply presto/dataset.py]: fixed

**Review by kvantricht — APPROVED** [review]
Looks good to me now!
